Entities/TDExplosive.h

```diff
@@ -1,332 +1,127 @@
+		/// </summary>
+		/// <param name="propName">The name of the property to be read.</param>
+		/// <param name="reader">A Reader lined up to the value of the property to be read.</param>
+		/// <returns>
+		/// An error return value signaling whether the property was successfully read or not.
+		/// 0 means it was read successfully, and any nonzero indicates that a property of that name could not be found in this or base classes.
+		/// </returns>
+		virtual int ReadProperty(std::string propName, Reader &reader);
+
+		/// <summary>
+		/// Saves the complete state of this TDExplosive to an output stream for later recreation with Create(Reader &reader);
+		/// </summary>
+		/// <param name="writer">A Writer that the TDExplosive will save itself with.</param>
+		/// <returns>An error return value signaling sucess or any particular failure. Anything below 0 is an error signal.</returns>
+		virtual int Save(Writer &writer) const;
+#pragma endregion
+
+#pragma region Destruction
+		/// <summary>
+		/// Destructor method used to clean up a TDExplosive object before deletion from system memory.
+		/// </summary>
+		virtual ~TDExplosive() { Destroy(true); }
+
+		/// <summary>
+		/// Destroys and resets (through Clear()) the SceneLayer object.
+		/// </summary>
+		/// <param name="notInherited">Whether to only destroy the members defined in this derived class, or to destroy all inherited members also.</param>
+		virtual void Destroy(bool notInherited = false);
+#pragma endregion
+
+#pragma region Virtual Override Methods
+		/// <summary>
+		/// Resets the entire TDExplosive, including its inherited members, to their default settings or values.
+		/// </summary>
+		virtual void Reset() { Clear(); ThrownDevice::Reset(); }
+
+		/// <summary>
+		/// Updates this MovableObject. Supposed to be done every frame.
+		/// </summary>
+		virtual void Update();
+
+		/// <summary>
+		/// Gets the ClassInfo instance of this Entity.
+		/// </summary>
+		/// <returns>A reference to the ClassInfo of this' class.</returns>
+		virtual const Entity::ClassInfo & GetClass() const { return m_sClass; }
+
+		/// <summary>
+		/// Gets the class name of this Entity.
+		/// </summary>
+		/// <returns>A string with the friendly-formatted type name of this object.</returns>
+		virtual const std::string & GetClassName() const { return m_sClass.GetName(); }
+
+		/// <summary>
+		/// Draws this' current graphical HUD overlay representation to a BITMAP of choice.
+		/// </summary>
+		/// <param name="pTargetBitmap">A pointer to a BITMAP to draw on.</param>
+		/// <param name="targetPos">The absolute position of the target bitmap's upper left corner in the Scene.</param>
+		/// <param name="whichScreen">Which player's screen this is being drawn to. May affect what HUD elements get drawn etc.</param>
+		/// <param name="playerControlled">Whether or not this MovableObject is currently player controlled (not applicable for TDExplosive)</param>
```

Comment:
Says right here the playerControlled parameter is not applicable to TDExplosive.
Should probably remove it from the method because it's that and not used.

Reply:
We can't really do this unfortunately, removing that'll change the method signature, so if anything tried to call DrawHUD on the explosive without a playerControlled entry it'd be calling some other class's DrawHUD instead, which is fucky.